worldedit-core/src/main/java/com/fastasyncworldedit/core/history/changeset/AbstractChangeSet.java

```diff
@@ -227,6 +230,28 @@ public synchronized IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet
         return set;
     }
 
+    private void processNegativeLayers(int bx, int bz, int layer, Object blocksGet, Object blocksSet) {
+        int by = layer << 4;
+        for (int y = 0, index = 0; y < 16; y++) {
+            int yy = y + by;
+            for (int z = 0; z < 16; z++) {
+                int zz = z + bz;
+                for (int x = 0; x < 16; x++, index++) {
+                    int xx = bx + x;
+                    int from = Array.getInt(blocksGet, index);
```

Comment:
This method is incredibly slow. That's why I approved the changes beforehand, as it's way harder to catch performance regressions if we try to clean up the code in the same run. I'd rather have duplicated code that can be cleaned up at any point than code that is cleaned up but needs a full rewrite to make it fast again.

Reply:
I shall revert that change. Would you prefer I defer making a helper class for the classes in the Blocks package like NotMyFault suggested to a separate pull request at a later date?

Reply:
I'm planning to not expose char/int arrays in IBlocks anymore, but I'll need some time for that (and it needs proper performance analysis). I therefore don't think you should spend too much time trying to deduplicate the code (and if you want, subsequent PRs are fine for me).

Reply:
As I have reverted this change as requested, I will go ahead and mark this as resolved.